### PR TITLE
Basic multiple files analysis

### PR DESCRIPTION
**Pull request by lukechurch**
@devoncarew PTAL

This adds the ability to analyse multiple dart files via a new EP 'analyzeMulti'. It doesn't link that through to the RPC APIs yet.

It cleans a bit more data betweens calls, but I don't think this will cause a measurable performance impact.

### CODE REVIEW

File: lib/src/analyzer.dart [hunk 1]
```diff
@@ -90,13 +105,65 @@ class Analyzer {
...
+      _context.applyChanges(changeSet);
+
+      List<AnalysisErrorInfo> errorInfos = [];
+      sourcesList.forEach((s) {
+        _context.computeErrors(s);
+        errorInfos.add(_context.getErrors(s));
+      });
+
+      List<_Error> errors = errorInfos
+      .expand((AnalysisErrorInfo info) {
```
Comment (devoncarew) [inline lib/src/analyzer.dart]: indent += 2

File: lib/src/api_classes.dart [hunk 2]
```diff
@@ -55,6 +57,24 @@ class SourceRequest {
...
+      description: 'Map of names to sources.')
+  Map<String, String> sources;
+
+  @ApiProperty(
+      description: 'An optional offset into the source code.')
+  int offset;
+
+  @ApiProperty(
+      description: 'Name of file that the offset is in')
+  String offsetName;
```
Comment (devoncarew) [inline lib/src/api_classes.dart]: Having the `offset` and `offsetName` be separate fields seems a little awkward. `offset` could be a `String`, `int` tuple?

**Comment by devoncarew** [comment]
lgtm, some comments

File: lib/src/api_classes.dart (hunk 2 above)
Comment (lukechurch) [inline lib/src/api_classes.dart]: I'll add a tiny 'Location' class to handle this.

File: lib/src/analyzer.dart (hunk 1 above)
Comment (lukechurch) [inline lib/src/analyzer.dart]: done

**Comment by lukechurch** [comment]
Thanks @devoncarew with simple tests I couldn't get a difference that was measurable. I'll try again with a larger test set, but it looks like it's going to be fine wrt performance.

I've addressed the review comments.

**Comment by lukechurch** [comment]
I will add an additional benchmark for multi-files in a separate PR.

**Comment by lukechurch** [comment]
@googlebot I am the author of these commits. The build break they caused is also entirely my fault.

**Pull request merged**